scaleway/resource_rdb_user_beta.go

```diff
@@ -0,0 +1,176 @@
+package scaleway
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/scaleway/scaleway-sdk-go/api/rdb/v1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func resourceScalewayRdbUserBeta() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceScalewayRdbUserBetaCreate,
+		Read:   resourceScalewayRdbUserBetaRead,
+		Update: resourceScalewayRdbUserBetaUpdate,
+		Delete: resourceScalewayRdbUserBetaDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"instance_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validationUUIDorUUIDWithLocality(),
+				Description:  "Instance on which the user is created",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Description: "Database user name",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"password": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Sensitive:   true,
+				Description: "Database user password",
+			},
+			"is_admin": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Grant admin permissions to database user",
+			},
+			// Common
+			"region": regionSchema(),
+		},
+	}
+}
+
+func resourceScalewayRdbUserBetaCreate(d *schema.ResourceData, m interface{}) error {
+	rdbAPI, region, err := rdbAPIWithRegion(d, m)
+	if err != nil {
+		return err
+	}
+	instanceId := d.Get("instance_id").(string)
+	createReq := &rdb.CreateUserRequest{
+		Region:     region,
+		InstanceID: expandID(instanceId),
+		Name:       d.Get("name").(string),
+		Password:   d.Get("password").(string),
+		IsAdmin:    d.Get("is_admin").(bool),
+	}
+
+	res, err := rdbAPI.CreateUser(createReq)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s/%s", region, expandID(instanceId), res.Name))
+
+	return resourceScalewayRdbUserBetaRead(d, m)
+}
+
+func resourceScalewayRdbUserBetaRead(d *schema.ResourceData, m interface{}) error {
+	rdbAPI, region, err := rdbAPIWithRegion(d, m)
+	if err != nil {
+		return err
+	}
+
+	regionName, instanceId, userName, err := resourceScalewayRdbUserBetaParseId(d.Id())
+
+	if err != nil {
+		return err
+	}
```

Comment:
`rdbAPIWithRegionAndID` doesn't support more than one / in the ID. Here I am storing Region/Instance/Username but this is not supported by the helpers it seems. Do you think I should modify the helpers to support this? This could have some side effects.